RecoParticleFlow/PFClusterProducer/plugins/alpaka/PFClusterSoAProducerKernel.dev.cc

```diff
@@ -0,0 +1,1628 @@
+          else if constexpr (debug)
+            printf("\tTopoId %d has diff = %f LESS than tolerance %f (terminating!)\n", topoId, diff, tol);
+        }
+      }
+      alpaka::syncBlockThreads(acc);  // all threads call sync
+    } while (notDone);                // shared variable ensures synchronization is well defined
+    if (once_per_block(acc))
+      for (int s = alpaka::getIdx<alpaka::Block, alpaka::Threads>(acc)[0u]; s < nSeeds; s += gridStride) {
+        int rhIdx = pfClusteringVars[s + pfClusteringVars[topoId].topoSeedOffsets()].topoSeedList();
+        int seedIdx = pfClusteringVars[rhIdx].rhIdxToSeedIdx();
+        clusterView[seedIdx].energy() = pfRecHits[s].energy();
+        clusterView[seedIdx].x() = pfRecHits[s].x();
+        clusterView[seedIdx].y() = pfRecHits[s].y();
+        clusterView[seedIdx].z() = pfRecHits[s].z();
+      }
+    alpaka::syncBlockThreads(acc);  // all threads call sync
+  }
+
+  template <typename TAcc, typename = std::enable_if_t<alpaka::isAccelerator<TAcc>>, bool debug = false>
+  ALPAKA_FN_ACC static void hcalFastCluster_multiSeedIterative(
+      const TAcc& acc,
+      reco::PFClusterParamsDeviceCollection::ConstView pfClusParams,
+      int topoId,
+      int nSeeds,
+      int nRHTopo,
+      reco::PFRecHitDeviceCollection::ConstView pfRecHits,
+      reco::PFClusteringVarsDeviceCollection::View pfClusteringVars,
+      reco::PFClusterDeviceCollection::View clusterView,
+      reco::PFRecHitFractionDeviceCollection::View fracView) {
+    int& nRHNotSeed = alpaka::declareSharedVar<int, __COUNTER__>(acc);
+    int& topoSeedBegin = alpaka::declareSharedVar<int, __COUNTER__>(acc);
+    int& gridStride = alpaka::declareSharedVar<int, __COUNTER__>(acc);
+    int& iter = alpaka::declareSharedVar<int, __COUNTER__>(acc);
+    float& tol = alpaka::declareSharedVar<float, __COUNTER__>(acc);
+    float& diff2 = alpaka::declareSharedVar<float, __COUNTER__>(acc);
+    float& rhENormInv = alpaka::declareSharedVar<float, __COUNTER__>(acc);
+    bool& notDone = alpaka::declareSharedVar<bool, __COUNTER__>(acc);
+
+    auto& clusterPos = alpaka::declareSharedVar<Position4[400], __COUNTER__>(acc);
+    auto& prevClusterPos = alpaka::declareSharedVar<Position4[400], __COUNTER__>(acc);
+    auto& clusterEnergy = alpaka::declareSharedVar<float[400], __COUNTER__>(acc);
+    auto& rhFracSum = alpaka::declareSharedVar<float[1500], __COUNTER__>(acc);
+    auto& seeds = alpaka::declareSharedVar<int[400], __COUNTER__>(acc);
+    auto& rechits = alpaka::declareSharedVar<int[1500], __COUNTER__>(acc);
+
+    if (once_per_block(acc)) {
+      nRHNotSeed = nRHTopo - nSeeds + 1;  // 1 + (# rechits per topoId that are NOT seeds)
+      topoSeedBegin = pfClusteringVars[topoId].topoSeedOffsets();
+      tol = pfClusParams.stoppingTolerance() *  // stopping tolerance * tolerance scaling
+            powf(fmaxf(1.0, nSeeds - 1.0), 2.0);
+      gridStride = alpaka::getWorkDiv<alpaka::Block, alpaka::Threads>(acc)[0u];
```

Comment:
This is called `gridStride`, but based on its definition and its usage later, it looks more like a block-level stride ?
Could you rename it to just `stride` ?